### PR TITLE
Add Grid button to Task Instance view

### LINKED ISSUE #39023: No way of going back to the DAG page once entering task details
### Apache Airflow version

2.9.0

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

Once entering the Task Details view, there's no way to directly go back to the DAG page.
image

In Airflow 2.8.4, we could still use the old navigation tabs (i.e `Grid`) to go back to the DAG page
image

### What you think should happen instead?

There should be a button to go back to the DAG page

### How to reproduce

1. Enter the `More Details` page of a TaskInstance.
2. Try to go back to the DAG page

### Operating System

Airflow in Kubernetes

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by csp33**
closes: #39023 

https://github.com/apache/airflow/assets/28935464/55e59e6c-41b8-44c7-b4b1-e20267a29237

### CODE REVIEW

**Comment by csp33** [comment]
@bbovenzi here's the PR ;)

File: airflow/www/templates/airflow/task_instance.html [hunk 1]
```diff
@@ -39,6 +39,9 @@ <h4>
         {{ url_for(endpoint, dag_id=dag.dag_id, task_id=task_id, execution_date=execution_date) }}
       {%- endif -%}
     {% endmacro -%}
+    <li><a href="{{ url_for('Airflow.grid', dag_id=dag.dag_id) }}">
```
Comment (bbovenzi) [inline airflow/www/templates/airflow/task_instance.html]: Nice! Could we include the task_id, dag_run_id, map_index in the params to link back to the same task instance?

Comment (csp33) [reply airflow/www/templates/airflow/task_instance.html]: I tried, but I don't know how to get the `dag_run_id` value 🤔 
Could you help me?

Comment (csp33) [reply airflow/www/templates/airflow/task_instance.html]: I did it! I'm not sure if it's the most optimal way, though.

Video has been updated in the PR description ;)

**Review by hussein-awala — APPROVED** [review]
Looks good

File: airflow/www/views.py [hunk 2]
```diff
@@ -1650,6 +1653,7 @@ def log(self, session: Session = NEW_SESSION):
             show_trigger_form_if_no_params=conf.getboolean("webserver", "show_trigger_form_if_no_params"),
             logs=logs,
             dag=dag_model,
+            dag_run_id=ti.dag_run.run_id if ti else "",
```
Comment (bbovenzi) [inline airflow/www/views.py]: We can just do `ti.run_id`

Comment (csp33) [reply airflow/www/views.py]: done!

**Review by bbovenzi — APPROVED** [review]
nice work!

**Pull request merged**